client/src/main/scala/org/apache/celeborn/client/CommitManager.scala

```diff
@@ -114,17 +114,23 @@ class CommitManager(appUniqueId: String, val conf: CelebornConf, lifecycleManage
 
                     if (workerToRequests.nonEmpty) {
                       val commitFilesFailedWorkers = new ShuffleFailedWorkers()
+                      val workerInfoSnapshots =
+                        lifecycleManager.shuffleAllocatedWorkers.get(shuffleId).keySet().asScala
+                          .map(worker => worker.toUniqueId() -> worker).toMap
+                      def getWorkerInfo(worker: WorkerInfo): WorkerInfo = {
+                        workerInfoSnapshots.get(worker.toUniqueId()).getOrElse(
+                          lifecycleManager.shuffleAllocatedWorkers
+                            .get(shuffleId)
+                            .asScala
+                            .find(_._1.equals(worker))
+                            .get
+                            ._1)
+                      }
                       try {
                         val params = new ArrayBuffer[CommitFilesParam](workerToRequests.size)
                         workerToRequests.foreach {
                           case (worker, requests) =>
-                            val workerInfo =
-                              lifecycleManager.shuffleAllocatedWorkers
-                                .get(shuffleId)
-                                .asScala
-                                .find(_._1.equals(worker))
-                                .get
-                                ._1
+                            val workerInfo = getWorkerInfo(worker)
```

Comment:
Could we use the hashcode of `WorkerInfo` directly?

Reply:
there might be hashcode conflicts.

And the` s"$host:$rpcPort:$pushPort:$fetchPort:$replicatePort"` is unique.

Reply:
AFAIK, there are a few hashmap using `WorkerInfo` as key.

Reply:
I used workerId as key in https://github.com/apache/celeborn/pull/2870

Reply:
> AFAIK, there are a few hashmap using `WorkerInfo` as key.

Here the key is workerId.
```
workerInfoSnapshots.get(worker.toUniqueId())
```

Reply:
Can we cache this for future commit requests? We need to ensure that it gets updated whenever the shuffle-related workers are modified.

Reply:
addressed. cc @RexXiong @onebox-li